plugins/scancode-consolidate-scan/src/plugin_consolidate/__init__.py

```diff
@@ -0,0 +1,435 @@
+                package_resource.extra_data['in_package_component'] = True
+                package_resource.save(codebase)
+
+                package_resource_license_expression = combine_expressions(package_resource.license_expressions)
+                package_resource_holders = package_resource.holders
+                if not package_resource_license_expression and not package_resource_holders:
+                    continue
+                discovered_license_expressions.append(package_resource_license_expression)
+                discovered_holders.extend(h.get('value') for h in package_resource_holders)
+
+            # Remove NoneTypes from discovered licenses
+            discovered_license_expressions = [lic for lic in discovered_license_expressions if lic]
+            # Remove NoneTypes from discovered holders
+            discovered_holders = [holder for holder in discovered_holders if holder]
+
+            combined_discovered_license_expression = combine_expressions(discovered_license_expressions)
+            if combined_discovered_license_expression:
+                simplified_discovered_license_expression = str(Licensing().parse(combined_discovered_license_expression).simplify())
+            else:
+                simplified_discovered_license_expression = None
+
+            c = Consolidation(
+                core_license_expression=package_license_expression,
+                other_license_expression=simplified_discovered_license_expression,
+                core_holders=sorted(set(package_holders)),
+                other_holders=sorted(set(discovered_holders)),
+                resources=package_resources,
+            )
+            yield ConsolidatedPackage(
+                package=package,
+                consolidation=c
+            )
+
+
+def get_license_holders_consolidated_components(codebase):
+    """
+    Yield a ConsolidatedComponent for each directory where 75% or more of the files have the
+    same license expression and copyright holders
+    """
+    # TODO: Create Consolidated Components for the 25% or less of files that
+    # aren't part of the majority
+    # TODO: Take license score into account
+    root = codebase.root
+    if root.extra_data.get('in_package_component'):
+        return
+
+    origin_translation_table = {}
+    for resource in codebase.walk(topdown=False):
+        # TODO: Consider facets for later
+
+        if resource.is_file or resource.extra_data.get('in_package_component'):
+            continue
+
+        children = resource.children(codebase)
+        if not children:
+            continue
+
+        # Collect license expression and holders count for stat-based summarization
+        origin_count = Counter()
+        for child in children:
+            if child.extra_data.get('in_package_component'):
+                continue
+            if child.is_file:
+                license_expression = combine_expressions(child.license_expressions)
```

Comment:
We need to add a reminder to consider the license match coverage and/or score here in the near future. Otherwise we may end up creating consolidated things that have licenses that were weakly detected.